lib/puppet/pops/types/p_sensitive_type.rb

```diff
@@ -0,0 +1,69 @@
+module Puppet::Pops
+module Types
+
+# A Puppet Language type that wraps sensitive information. The sensitive type is parameterized by
+# the wrapped value type.
+#
+#
+# @api public
+class PSensitiveType < Puppet::Pops::Types::PScalarType
+
+  class Sensitive
+    def initialize(value)
+      @value = value
+    end
+
+    def unwrap
+      @value
+    end
+
+    def to_s
+      "Sensitive"
+    end
+
+    def inspect
+      "#<Sensitive>"
+    end
+  end
+
+  attr_reader :element_type
+
+  def initialize(element_type = Object)
+    @element_type = element_type
+  end
+
+  def instance?(o, guard = nil)
+    o.is_a?(Sensitive)
+  end
+
+  def eql?(o)
+    self.class == o.class
+  end
+
+  def generalize
+    DEFAULT
+  end
+
+  def self.new_function(_, loader)
+    @new_function ||= Puppet::Functions.create_loaded_function(:new_Sensitive, loader) do
+      dispatch :new do
```

Comment:
Probably also want to unwrap if given a Sensitive value. That is, if users does something like this:
```
$a = Sensitive(42)
$b = Sensitive($a)
```
Is the type of $b `Sensitive[Sensitive[Integer]]`? That seems nonsensical. Thus, a dispatcher for `Sensitive[Any]` would simply return the given value since it is already a `Sensitive` instance.